java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -738,15 +738,49 @@ public ColumnBuilder appendLists(List... inputLists) {
     public ColumnBuilder appendStructValues(List<StructData> inputList) {
       for (StructData structInput : inputList) {
         // one row
-        rows++;
         append(structInput);
       }
       return this;
     }
 
+    private void resizeDataBuffer(int length) {
+      // just for strings we want to throw a real exception if we would overrun the buffer
+      if (data == null) {
+        // since rows can be updated as we go, make data buffer as big as the estimate or the requested length
+        if (!type.isNestedType() && type != DType.STRING) {
+          data = HostMemoryBuffer.allocate(estimatedRows * type.getSizeInBytes());
+        } else {
+          data = HostMemoryBuffer.allocate(length);
+        }
+        return;
+      }
+      long oldLen = data.getLength();
+      long newLen = Math.max(1, oldLen);
+      while (currentByteIndex + length > newLen) {
+        newLen *= 2;
+      }
+      if (newLen > Integer.MAX_VALUE && type == DType.STRING) {
+        throw new IllegalStateException("A string buffer is not supported over 2GB in size");
```

Comment:
@revans2 , do we have the max value limit for data buffers for other datatypes than String as well? I am guessing yes, in which case I will have to fix this piece a bit but wanted to confirm what was expected here.

Reply:
Shouldn't this be max number of rows* type size if it is not a string. Pardon any naivety here

Reply:
Yes, all columns are indexed by a 32-bit int, so if you need to stay within that range. This goes for child columns too.

Reply:
That's what I thought but I'm still not clear about what the length of a data buffer of ints column with row count = int max value would be? Appreciate the help.

Reply:
it is always row count.  The string data column just happens to be bytes so it is the same.

Reply:
Thank you. Agree that the index limit /bye indices would be int max. I might have to revisit it here since I may have missed the point here. To me, how do you fit ints with row count =int max in a data buffer of int max limit where every entry is say 4 bytes long. Sorry about the back and forth here, is appreciate the explanation.

Reply:
Ok, so this is what was throwing me off and I am sharing what was going on IMO. In the old builder, the resize happened only if the type was string, which meant we checked if were over 2 GB limit only in this case. If the estimated rows to the old builder for a fixed width column was such that the datatype size * that row count was > 2 GB we did not assert it when initing the data buffer. As part of the latest change I am limiting it to 2 GB no matter the type, let me know if that is ok and we can work with it some more. The way one can see the difference in behavior is by running plugin tests where   estimatedRows=268435456 and we are writing longs.
```- Write doubles to long (values within range) *** FAILED ***```